backfill.py

```diff
@@ -130,27 +146,165 @@ def get_result_from_durations(start_up_durations_path, analyzed_path):
+
+    if checkout_proc.returncode != 0:
+        print(("\n\nSomething went wrong while checking out this commit: {commit} . The associated error message was:"
+               "\n\n {error}".format(commit=hash, error=checkout_proc.stderr.decode('utf-8').strip("\n"))),
+              file=sys.stderr)
+    elif assemble_proc.returncode != 0:
+        print(("\n\nSomething went wrong while assembling this build: {build} . The associated error message was:"
+               "\n\n {error}".format(build=build_type, error=checkout_proc.stderr.decode('utf-8').strip("\n"))),
+              file=sys.stderr)
+
+
+def build_apk_path_string(repository_path, build_type, phone_architecture):
+    build_apk_destination = repository_path + "/app/build/outputs/apk/" + build_type + "/"
+    apk_name = "app-" + phone_architecture + "-" + build_type + ".apk"
+    return build_apk_destination + apk_name
+
+
+def move_apk_to_cwd(apk_path, commit_hash):
+    new_apk_name = "apk_commit_" + commit_hash + ".apk"
+    proc = subprocess.run(["mv", apk_path, new_apk_name])
+    if proc.returncode != 0:
+        print(("\n\nSomething went wrong while moving the built apk: {apk} . The associated error message was:"
+               "\n\n {error}".format(apk=apk_path, error=proc.stderr.decode('utf-8').strip("\n"))),
+              file=sys.stderr)
+    return new_apk_name
+
+
+def build_apks_for_commits(
+        startdate=None, enddate=None, start_commit=None, end_commit=None, repository_path=None,
+        build_type=None, architecture=None, branch="", remote_name=""):
+    apk_metadata_array = []
+
+    checkout_repository_to_correct_branch(repository_path, branch, remote_name)
+
+    if startdate is None:
+        array_of_commit_hash = get_all_commits_in_commits_range(start_commit, end_commit, repository_path)
+    else:
+        array_of_commit_hash = get_commits_for_date_range(startdate, enddate, repository_path)
+    for commit in array_of_commit_hash:
+        build_apk_for_commit(commit, repository_path, build_type)
+        built_apk_name = build_apk_path_string(repository_path, build_type, architecture)
+        new_apk_name = move_apk_to_cwd(built_apk_name, commit)
+        apk_metadata_array.append({
+            KEY_NAME: new_apk_name,
+            KEY_DATETIME: "",
+            KEY_COMMIT: commit,
+            KEY_ARCHITECTURE: architecture})
+    return apk_metadata_array
 
 
 def cleanup(array_of_apk_path):
     for i in array_of_apk_path:
         subprocess.run(["rm", i[KEY_NAME]])
 
 
+def validate_args(args):
+    if args.type == "commitsDate" and not args.startdate:
+        raise Exception("Running backfill with commits between two date requires a start date")
+    if args.type == "commitsRange" and not args.startcommit and not args.endcommit:
+        raise Exception("Running backfill with commits between two commits requires a start and end commit")
+    if (args.type == "commitsRange" or args.type == "commitsDate") and not args.repository_path:
```

Comment:
nitpick (blocking): we're asking the user to pass in the repo twice: once as a positional argument (path to start up script) and once with `repository_path`: let's simplify it and just ask for a path to the repo, where we can construct the path to the script.